proxy/handlers/handlers.go

```diff
@@ -24,16 +24,19 @@ type Handlers struct {
 	Destinations destinations.Destinations
 	IgnoreTags   []matcher.TagMatcher
 	Logger       *logrus.Entry
+	Shutdown     bool
 	Statsd       scopedstatsd.Client
 }
 
 func (proxy *Handlers) HandleHealthcheck(
 	writer http.ResponseWriter, request *http.Request,
 ) {
-	if proxy.Destinations.Size() > 0 {
-		writer.WriteHeader(http.StatusNoContent)
-	} else {
+	if proxy.Shutdown {
+		writer.WriteHeader(http.StatusNotFound)
```

Comment:
Does a 410 status code make more semantic sense here?

Reply:
I looked this up, and it seems like Service Unavailable (503) is what is recommended.